pulsar-client-cpp/CMakeLists.txt

```diff
@@ -345,10 +345,15 @@ include_directories(
   ${CURL_INCLUDE_DIRS}
   ${Protobuf_INCLUDE_DIRS}
   ${LOG4CXX_INCLUDE_PATH}
-  ${GTEST_INCLUDE_PATH}
-  ${GMOCK_INCLUDE_PATH}
```

Comment:
Even if these variables are not set, it won't make build failed.

Actually there are already a CI to verify the case when `BUILD_TESTS=OFF` without GTest dependency. See

https://github.com/apache/pulsar/blob/f0aeb5acfb127c42d8f5dda65804617e69e77dd3/.github/workflows/ci-cpp-build.yaml#L231-L234

https://github.com/apache/pulsar/blob/f0aeb5acfb127c42d8f5dda65804617e69e77dd3/pulsar-client-cpp/pkg/deb/docker-build-deb.sh#L33-L34

https://github.com/apache/pulsar/blob/f0aeb5acfb127c42d8f5dda65804617e69e77dd3/pulsar-client-cpp/pkg/deb/build-deb.sh#L44

Your CMake option is wrong, you should use `-DBUILD_TESTS=OFF`, not `-DTESTS=OFF`.

Reply:
Sorry for the typo. It is `cmake . -DBUILD_TESTS=OFF` indeed. I've corrected it.
![image](https://user-images.githubusercontent.com/16649269/192221579-6c2213a6-92cb-4670-a3be-4f9f4596b752.png)